configure.ml

```diff
@@ -431,6 +431,20 @@ let vcs =
   else if dir_exists "{arch}" then "gnuarch"
   else "none"
 
+(** * Git Precommit Hook *)
+let _ =
+  let f = ".git/hooks/pre-commit" in
+  if vcs = "git" && dir_exists ".git/hooks" && not (Sys.file_exists f) then begin
+    printf "Creating pre-commit hook in %s\n" f;
+    let o = open_out f in
+    let pr s = fprintf o s in
+    pr "#!/bin/sh\n";
+    pr "\n";
+    pr "test -x dev/tools/pre-commit && . dev/tools/pre-commit\n";
```

Comment:
If dev/tools/pre-commit doesn't exist this will return an error code. Use an `if` instead.

Reply:
Also we should execute, not source so that if we ever use a different shebang (ie want to use bash in the hook) we don't have to fix things.